src/ScriptCs.Engine.Mono/Segmenter/ScriptSegmenter.cs

```diff
@@ -21,54 +21,56 @@ public List<SegmentResult> Segment(string code)
-                        BeginLine = lineNr,
-                        Code = segment
-                    });
-                }
-                else
+                foreach (var region in parser.Parse(code))
                 {
-                    var isMethod = analyser.IsMethod(segment);
+                    // Calculate region linenumber
+                    var lineNr = code.Substring(0, region.Offset).Count(x => x.Equals('\n'));
 
-                    if (isMethod)
-                    {
-                        // method ok
-                        var method = analyser.ExtractPrototypeAndMethod(segment);
-
-                        result.Add(new SegmentResult
-                        {
-                            Type = SegmentType.Prototype,
-                            BeginLine = lineNr,
-                            Code = method.ProtoType
-                        });
+                    var segment = code.Substring(region.Offset, region.Length);
 
+                    if (analyser.IsClass(segment))
+                    {
                         result.Add(new SegmentResult
                         {
-                            Type = SegmentType.Method,
+                            Type = SegmentType.Class,
                             BeginLine = lineNr,
-                            Code = method.MethodExpression
+                            Code = segment
                         });
                     }
                     else
                     {
-                        result.Add(new SegmentResult
+                        var isMethod = analyser.IsMethod(segment);
+
+                        if (isMethod)
                         {
-                            Type = SegmentType.Evaluation,
-                            BeginLine = lineNr,
-                            Code = segment
-                        });
+                            // method ok
+                            var method = analyser.ExtractPrototypeAndMethod(segment);
+
+                            result.Add(new SegmentResult
+                            {
+                                Type = SegmentType.Prototype,
+                                BeginLine = lineNr,
+                                Code = method.ProtoType
+                            });
+
+                            result.Add(new SegmentResult
+                            {
+                                Type = SegmentType.Method,
+                                BeginLine = lineNr,
+                                Code = method.MethodExpression
+                            });
+                        }
+                        else
+                        {
+                            result.Add(new SegmentResult
+                            {
+                                Type = SegmentType.Evaluation,
+                                BeginLine = lineNr,
+                                Code = segment
+                            });
+                        }
```

Comment:
This is actually a very trivial change, look at it via https://github.com/scriptcs/scriptcs/pull/1006/files?w=0#diff-fcd5dd5eb14161d870a7e175c544c1a4